utility_analysis_new/combiners.py

```diff
@@ -0,0 +1,135 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""UtilityAnalysisCountCombiner."""
+
+from dataclasses import dataclass
+from typing import List, Sized
+import numpy as np
+
+import pipeline_dp
+from pipeline_dp import dp_computations
+
+
+@dataclass
+class CountUtilityAnalysisMetrics:
+    """ Stores metrics for the count utility analysis.
+
+    Args:
+        count: the number of contributions per a partition.
+        per_partition_contribution_error: the amount of contribution error in a partition.
+        expected_cross_partition_error: the estimated amount of error across partitions.
+        var_cross_partition_error: the variance of the contribution bounding error.
+        std_cb: the standard deviation of the var_cross_partition_error.
+        std_noise: the noise standard deviation for DP count.
+        noise_kind: the type of noise used.
+    """
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    var_cross_partition_error: float
+    std_cb: float
+    std_noise: float
+    noise_kind: pipeline_dp.NoiseKind
+
+
+@dataclass
+class AccumulatorMetrics:
+    count: int
+    per_partition_contribution_error: int
+    expected_cross_partition_error: float
+    var_cross_partition_error: float
+
+
+class UtilityAnalysisCountCombiner(pipeline_dp.Combiner):
+    """ A combiner for utility analysis counts. """
+    AccumulatorType = AccumulatorMetrics
+
+    def __init__(self, params: pipeline_dp.combiners.CombinerParams):
+        self._params = params
+
+    def create_accumulator(self, data: Sized) -> AccumulatorType:
+        """ Create an accumulator for a user and a partition.
+
+        Args:
+            data: contains two columns; 1) a column with the total number of partitions a user contributed to
+            and 2) a column with a list of the user's contributions for a single partition.
+
+        Returns:
+            An accumulator with the count of contributions and the contribution error for this partition, and the
+            estimated error and variance across all of the partitions for this user.
+        """
+        if not data:
+            return AccumulatorMetrics(0, 0, 0, 0)
+        n_partitions, values = data
+        count = len(values)
+        max_per_partition = self._params.aggregate_params.max_contributions_per_partition
+        max_partitions = self._params.aggregate_params.max_partitions_contributed
+        per_partition_contribution_error = max(0, count - max_per_partition)
+        prob_keep_partition = min(1, max_partitions /
+                                  n_partitions) if n_partitions > 0 else 0
+        expected_cross_partition_error = per_partition_contribution_error * prob_keep_partition
```

Comment:
Cross partition error comes from dropping contributions because of cross-partition contribution bounding:
in cross-partition conribution bounding, there are 2 options:
 ---    `0`, with probablity (1-`prob_keep_partition`)
 ---    `min(max_per_partition, count)` with probability `prob_keep_partition`

In first case error (i.e. dropped amount) is equal to `min(max_per_partition, count)`, in the 2nd no error intoduced.
So expected_cross_partition_error = `min(max_per_partition, count)*(1-prob_keep_partition)`.

we can set:
`per_partition_contribution = min(max_per_partition, count)`
and then 
```
expected_cross_partition_error = `per_partition_contribution*(1-prob_keep_partition).
var_cross_partition_error = per_partition_contribution**2 * prob_keep_partition * (1 - prob_keep_partition)

```

Reply:
Thank you for the detailed explanation! I've modified the calculations and the tests; it was backwards before (: 🚀 PTAL thank you!